templatefuncs.go

```diff
@@ -159,6 +175,62 @@ func regexpReplaceAllTemplateFunc(expr, repl, s string) string {
 	return regexp.MustCompile(expr).ReplaceAllString(s, repl)
 }
 
+// reverseTemplateFunc is the core implementation of the `reverse`
+// template function.
+func reverseTemplateFunc(list []any) []any {
+	listcopy := append([]any(nil), list...)
+	slices.Reverse(listcopy)
+	return listcopy
+}
+
+// sortTemplateFunc is the core implementation of the `sort` template function.
+func sortTemplateFunc(list []any) any {
```

Comment:
Trying to write a generic sort function is hard. I would instead have separate `sortInts` and `sortFloat64s` functions, with `sort` converting all arguments to `string`s and then sorting those.

Reply:
Is what I described as [terrible sort](https://github.com/chezmoi/templatefuncs/pull/29/commits/b674fe912de30997bf42610159217d839e5dd754#diff-6b65c8a16c3fb82c4c27546f0b99c1d4e7904503c415c69efeb28574e565a0d4R183-R201) roughly sufficient?